### PR TITLE
Add function for a flush thread with custom interval

### PR DESCRIPTION
**Pull request by res0nance**
Allows users to customize the behaviour runtime.

### CODE REVIEW

File: zlog.c [hunk 1]
```diff
@@ -100,21 +100,23 @@ void zlog_init_stderr(void)
     zlog_fout = stderr;
 }
 
-static void* zlog_buffer_flush_thread()
+static void* zlog_buffer_flush_thread(void* interval_sec)
 {
     struct timeval tv;
     time_t lasttime;
     time_t curtime;
+    int flush_interval = (int)(intptr_t)interval_sec;
```
Comment (zma) [inline zlog.c]: `*(intptr_t)interval_sec`?

File: zlog.c [hunk 2]
```diff
@@ -135,9 +137,14 @@ static void* zlog_buffer_flush_thread()
...
 void zlog_init_flush_thread(void)
+{
+    zlog_init_flush_thread_with_interval(ZLOG_FLUSH_INTERVAL_SEC);
+}
+
+void zlog_init_flush_thread_with_interval(int interval_sec)
 {
     pthread_t thr;
-    pthread_create(&thr, NULL, zlog_buffer_flush_thread, NULL);
+    pthread_create(&thr, NULL, zlog_buffer_flush_thread, (void*)(intptr_t)interval_sec);
```
Comment (zma) [inline zlog.c]: `malloc` from heap for storing the `arg` passed to `pthread_create` will be safer.

**Review by zma — CHANGES_REQUESTED** [review]

**Comment by zma** [comment]
Thanks @res0nance for the PR! Please find comments above.

**Comment by res0nance** [comment]
> Thanks @res0nance for the PR! Please find comments above.

Hey @zma thanks for the review, I've addressed the comments

**Pull request merged**